System.IO.Abstractions.TestingHelpers.Tests/MockFileCopyTests.cs

```diff
@@ -27,6 +27,27 @@ public void MockFile_Copy_ShouldOverwriteFileWhenOverwriteFlagIsTrue()
             Assert.AreEqual(copyResult.Contents, sourceContents.Contents);
         }
 
+        [Test]
+        public void MockFile_Copy_ShouldCloneContents()
+        {
+            var sourceFileName = XFS.Path(@"c:\source\demo.txt");
+            var destFileName = XFS.Path(@"c:\source\demo_copy.txt");
+
+            var mockFileSystem = new MockFileSystem();
+            mockFileSystem.AddFile(sourceFileName, "Original");
+            mockFileSystem.File.Copy(sourceFileName, destFileName);
+            
+            using (var stream = mockFileSystem.File.Open(sourceFileName, FileMode.Open, FileAccess.ReadWrite))
```

Comment:
I'd probably use a more direct approach to modifying the file. Straight up deleting it, clearing its contents, etc. Seems a little overkill to leverage a binarywriter, and set the contents to modified.

Reply:
Hm, for a repro the byte array holding the contents needs to be directly modified. I think most other APIs will go through the string path. Will check later if there's something simpler that'l repro.

Reply:
I didn't actually mess around with it, but looking at the issue, it sounds like the copied file is actually just a reference to the original. To me that means if we were to say.. delete the original, the copied file would be deleted as well (because of the found bug).

Reply:
No, deleting the old file has no effect, because this just removes the old entry from the dictionary (without touching the data). This behavior is only observable when using `MockFileStream` to modify the contents of the file directly.